### ISSUE TITLE
TSQL Alter Table unparsable error for DROP PERIOD

### ISSUE THREAD

**Issue by nathanthorell** [sqlfluff/sqlfluff#6585]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQL Statement:
```sql
ALTER TABLE TestTable DROP PERIOD FOR SYSTEM_TIME;
```
Throws an unparsable error.

### Expected Behaviour

This is valid t-sql as it relates to TEMPORAL System-versioned tables:
https://learn.microsoft.com/en-us/sql/relational-databases/tables/stopping-system-versioning-on-a-system-versioned-temporal-table

It is also listed in the ALTER TABLE statement:
https://learn.microsoft.com/en-us/sql/t-sql/statements/alter-table-transact-sql

I expect the above test statement should lint without issue.

### Observed Behaviour

Found unparsable section: 'ALTER TABLE TestTable DROP PERIOD FOR SY...'

### How to reproduce

test.sql
```sql
ALTER TABLE TestTable DROP PERIOD FOR SYSTEM_TIME;
```

Then run:
```bash
sqlfluff lint test.sql --dialect=tsql
```

### Dialect

t-sql

### Version

sqlfluff, version 3.3.0
Python 3.12.8

### Configuration

[sqlfluff]
dialect = tsql
templater = jinja

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
